File: Backend/Application/Services/DashboardService.cs
using Application.Commons.Bases.Response;
using Application.Dtos.Response.Dashboard;
using Application.Interfaces;
using Application.Mappers;
using Infrastructure.Persistences.Interfaces;
using Utilities.Static;

namespace Application.Services
{
    public class DashboardService : IDashboardService
    {
        private readonly IUnitOfWork _unitOfWork;

        public DashboardService(IUnitOfWork unitOfWork)
        {
            _unitOfWork = unitOfWork;
        }

        public async Task<BaseResponse<DashboardGoodsIssueStatsResponseDto>> GetGoodsIssueStats(int authenticatedStoreId, CancellationToken cancellationToken)
        {
            var response = new BaseResponse<DashboardGoodsIssueStatsResponseDto>();

            try
            {
                var stats = await _unitOfWork.DashboardQuery.GetGoodsIssueStatsLast14DaysAsync(authenticatedStoreId, cancellationToken);
                response.IsSuccess = true;
                response.Data = DashboardMapp.GoodsIssueStatsResponseDtoMapping(stats);
                response.Message = ReplyMessage.MESSAGE_QUERY;
            }
            catch (Exception ex)
            {
                response.IsSuccess = false;
                response.Message = ReplyMessage.MESSAGE_EXCEPTION + ex.Message;
            }

            return response;
        }

        public async Task<BaseResponse<DashboardStoreInventoryStatsResponseDto>> GetInventoryStats(int authenticatedStoreId, CancellationToken cancellationToken)
        {
            var response = new BaseResponse<DashboardStoreInventoryStatsResponseDto>();

            try
            {
                var stats = await _unitOfWork.DashboardQuery.GetInventoryStatsCurrentMonthAsync(authenticatedStoreId, cancellationToken);
                response.IsSuccess = true;
                response.Data = DashboardMapp.InventoryStatsResponseDtoMapping(stats);
                response.Message = ReplyMessage.MESSAGE_QUERY;
            }
            catch (Exception ex)
            {
                response.IsSuccess = false;
                response.Message = ReplyMessage.MESSAGE_EXCEPTION + ex.Message;
            }

            return response;
        }

        public async Task<BaseResponse<IEnumerable<DashboardMovementResponseDto>>> GetMovementsStats(int authenticatedStoreId, CancellationToken cancellationToken)
        {
            var response = new BaseResponse<IEnumerable<DashboardMovementResponseDto>>();

            try
            {
                var movements = await _unitOfWork.DashboardQuery.GetMovementsLast6MonthsAsync(authenticatedStoreId, cancellationToken);
                response.IsSuccess = true;
                response.Data = movements.Select(DashboardMapp.MovementStatsResponseDtoMapping);
                response.Message = ReplyMessage.MESSAGE_QUERY;
            }
            catch (Exception ex)
            {
                response.IsSuccess = false;
                response.Message = ReplyMessage.MESSAGE_EXCEPTION + ex.Message;
            }

            return response;
        }

        public async Task<BaseResponse<DashboardProductReplenishmentResponseDto>> GetProductReplenishment(int authenticatedStoreId, CancellationToken cancellationToken)
        {
            var response = new BaseResponse<DashboardProductReplenishmentResponseDto>();

            try
            {
                var status = await _unitOfWork.DashboardQuery.GetProductReplenishmentAsync(authenticatedStoreId, cancellationToken);
                response.IsSuccess = true;
                response.Data = DashboardMapp.ProductReplenishmentResponseDtoMapping(status);
                response.Message = ReplyMessage.MESSAGE_QUERY;
            }
            catch (Exception ex)
            {
                response.IsSuccess = false;
                response.Message = ReplyMessage.MESSAGE_EXCEPTION + ex.Message;
            }

            return response;
        }

        public async Task<BaseResponse<DashboardProductStatsResponseDto>> GetProductStats(CancellationToken cancellationToken)
        {
            var response = new BaseResponse<DashboardProductStatsResponseDto>();

            try
            {
                var stats = await _unitOfWork.DashboardQuery.GetProductStatsCurrentMonthAsync(cancellationToken);
                response.IsSuccess = true;
                response.Data = DashboardMapp.ProductStatsResponseDtoMapping(stats);
                response.Message = ReplyMessage.MESSAGE_QUERY;
            }
            catch (Exception ex)
            {
                response.IsSuccess = false;
                response.Message = ReplyMessage.MESSAGE_EXCEPTION + ex.Message;
            }

            return response;
        }

        public async Task<BaseResponse<IEnumerable<DashboardTransferByStoreResponseDto>>> GetTransfersByStore(int authenticatedStoreId, CancellationToken cancellationToken)
        {
            var response = new BaseResponse<IEnumerable<DashboardTransferByStoreResponseDto>>();
            try
            {
                var transfers = await _unitOfWork.DashboardQuery.GetTransfersByStoreLast6MonthsAsync(authenticatedStoreId, cancellationToken);
                response.IsSuccess = true;
                response.Data = transfers.Select(DashboardMapp.TransferByStoreResponseDtoMapping);
                response.Message = ReplyMessage.MESSAGE_QUERY;
            }
            catch (Exception ex)
            {
                response.IsSuccess = false;
                response.Message = ReplyMessage.MESSAGE_EXCEPTION + ex.Message;
            }
            return response;
        }

        public async Task<BaseResponse<DashboardTransferStatsResponseDto>> GetTransferPending(int authenticatedStoreId, CancellationToken cancellationToken)
        {
            var response = new BaseResponse<DashboardTransferStatsResponseDto>();

            try
            {
                var stats = await _unitOfWork.DashboardQuery.GetTransferPendingAsync(authenticatedStoreId, cancellationToken);
                response.IsSuccess = true;
                response.Data = DashboardMapp.TransferPendingResponseDtoMapping(stats);
                response.Message = ReplyMessage.MESSAGE_QUERY;
            }
            catch (Exception ex)
            {
                response.IsSuccess = false;
                response.Message = ReplyMessage.MESSAGE_EXCEPTION + ex.Message;
            }

            return response;
        }

        public async Task<BaseResponse<IEnumerable<DashboardTransferStatusResponseDto>>> GetTransferStatus(int authenticatedStoreId, CancellationToken cancellationToken)
        {
            var response = new BaseResponse<IEnumerable<DashboardTransferStatusResponseDto>>();

            try
            {
                var transfers = await _unitOfWork.DashboardQuery.GetTransferStatusLast6MonthsAsync(authenticatedStoreId, cancellationToken);
                response.IsSuccess = true;
                response.Data = transfers.Select(DashboardMapp.TransferStatusResponseDtoMapping);
                response.Message = ReplyMessage.MESSAGE_QUERY;
            }
            catch (Exception ex)
            {
                response.IsSuccess = false;
                response.Message = ReplyMessage.MESSAGE_EXCEPTION + ex.Message;
            }

            return response;
        }
    }
}

File: Frontend/src/services/dashboardService.ts
import axios from 'axios';
import { BaseResponse } from '@/interfaces/baseInterface';
import { 
  DashboardGoodsIssueStats, 
  DashboardInventoryStats,
  DashboardMovementStats,
  DashboardProductReplenishment, 
  DashboardProductStats,
  DashboardTransferByStore, 
  DashboardTransferPending,
  DashboardTransferStatus } from '@/interfaces/dashboardInterface';

class DashboardService {
  async getGoodsIssueStats(): Promise<DashboardGoodsIssueStats> {
    const response = await axios.get<BaseResponse<DashboardGoodsIssueStats>>('api/Dashboard/GoodsIssueStats');
    return response.data.data;
  }

  async getInventoryStats(): Promise<DashboardInventoryStats> {
    const response = await axios.get<BaseResponse<DashboardInventoryStats>>('api/Dashboard/InventoryStats');
    return response.data.data;
  }

  async getMovementsStats(): Promise<DashboardMovementStats[]> {
    const response = await axios.get<BaseResponse<DashboardMovementStats[]>>('api/Dashboard/MovementsStats');
    return response.data.data;
  }

  async getProductReplenishment(): Promise<DashboardProductReplenishment> {
    const response = await axios.get<BaseResponse<DashboardProductReplenishment>>('api/Dashboard/ProductReplenishment');
    return response.data.data;
  }

  async getProductStats(): Promise<DashboardProductStats> {
    const response = await axios.get<BaseResponse<DashboardProductStats>>('api/Dashboard/ProductStats');
    return response.data.data;
  }

  async getTransfersByStore(): Promise<DashboardTransferByStore[]> {
    const response = await axios.get<BaseResponse<DashboardTransferByStore[]>>('api/Dashboard/TransfersByStore');
    return response.data.data;
  }

  async getTransferPending(): Promise<DashboardTransferPending> {
    const response = await axios.get<BaseResponse<DashboardTransferPending>>('api/Dashboard/TransferPending');
    return response.data.data;
  }

  async getTransferStatus(): Promise<DashboardTransferStatus[]> {
    const response = await axios.get<BaseResponse<DashboardTransferStatus[]>>('api/Dashboard/TransferStatus');
    return response.data.data;
  }
}

export const dashboardService = new DashboardService();